modifiers/heading_id.test.js

```diff
@@ -119,6 +119,42 @@ const mdSluggify = require('markdown-it')().use(require('./heading_id'), {
 });
 
 it('supports a custom sluggify function', () => {
-    expect(mdSluggify.render('# Hello World!')).toBe(`<h1 id="helloworld">Hello World!</h1>
+    expect(mdSluggify.render('# Hello World!')).toBe(`<h1 id="helloworld"><a class="hash-anchor" href="#helloworld" aria-hidden="true"></a>Hello World!</h1>
+`);
+});
+
+const mdDisableHashLink = require('markdown-it')().use(require('./heading_id'), {
+    hashLink: false,
+});
+
+it('supports disabling hash links feature', () => {
+    expect(mdDisableHashLink.render('# Hello World!')).toBe(`<h1 id="hello-world">Hello World!</h1>
+`);
+});
+
+const mdMaxHashLinkLevel = require('markdown-it')().use(require('./heading_id'), {
+    hashLink: {
+        maxLevel: 5,
+    },
+});
+
+it('supports adjusting what level of headings generate hash links', () => {
+    expect(mdMaxHashLinkLevel.render('##### Hello World!')).toBe(`<h5 id="hello-world"><a class="hash-anchor" href="#hello-world" aria-hidden="true"></a>Hello World!</h5>
+`);
+});
+
+const mdCustomAnchorClass = require('markdown-it')().use(require('./heading_id'), {
+    hashLink: {
+        class: 'custom-anchor',
+    },
+});
+
+it('supports changing class name for anchor in hash link', () => {
+    expect(mdCustomAnchorClass.render('# Hello World!')).toBe(`<h1 id="hello-world"><a class="custom-anchor" href="#hello-world" aria-hidden="true"></a>Hello World!</h1>
+`);
+});
+
+it('supports rendering heading without anchor above maxLevel', () => {
+    expect(md.render('#### Hello World!')).toBe(`<h4 id="hello-world">Hello World!</h4>
 `);
 });
```

Comment:
Can this test be moved above the tests that use new instances? (so after `resets exposed headings between repeat renders`)